cpp/collections_module/algorithm/collections.cpp

```diff
@@ -50,59 +83,56 @@ void Collections::Avg(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *res
     }
     average /= list.Size();
 
-    auto record = record_factory.NewRecord();
-    record.Insert(std::string(kResultAvg).c_str(), average);
+    result.SetValue(average);
 
   } catch (const std::exception &e) {
-    record_factory.SetErrorMessage(e.what());
+    result.SetErrorMessage(e.what());
     return;
   }
 }
 
-void Collections::ContainsAll(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+void Collections::ContainsAll(mgp_list *args, mgp_func_context *ctx, mgp_func_result *res, mgp_memory *memory) {
   mgp::memory = memory;
   const auto arguments = mgp::List(args);
-  const auto record_factory = mgp::RecordFactory(result);
+  auto result = mgp::Result(res);
 
   try {
     const auto list1{arguments[0].ValueList()};
     std::unordered_set<mgp::Value> set;
 
-    for (const auto &elem: list1){
+    for (const auto &elem : list1) {
       set.insert(elem);
     }
     const auto list2{arguments[1].ValueList()};
     std::unordered_set<mgp::Value> values;
 
-    for (const auto &elem: list2){
+    for (const auto &elem : list2) {
       values.insert(elem);
     }
-    auto record = record_factory.NewRecord();
-    record.Insert(std::string(kResultContainsAll).c_str(),
-                  std::all_of(values.begin(), values.end(), [&](const auto &x) { return set.contains(x); }));
+    result.SetValue(std::all_of(values.begin(), values.end(), [&](const auto &x) { return set.contains(x); }));
```

Comment:
```suggestion
    result.SetValue(std::all_of(values.begin(), values.end(), [&set](const auto &x) { return set.contains(x); }));
```
It is suggested that you don't use default capture (reference capture), or at least that we use reference capture by variables we need.

So I would recommend in this case changing it like this, in other cases you can check in memgraph that we are using init capture: https://stackoverflow.com/questions/25408190/lambda-captures-c14